.github/workflows/ci_linux_x64_clang_asan.yml

```diff
@@ -12,6 +12,7 @@ on:
       - main
     paths:
       - ".github/workflows/ci_linux_x64_clang_asan.yml"
```

Comment:
Uhhhh something in here isn't triggering how I'd like? https://github.com/iree-org/iree/actions/workflows/ci_linux_x64_clang_asan.yml should have way more runs. That might explain why @hanhanW had seemingly unrelated failures on the LLVM integrate: https://github.com/iree-org/iree/pull/18102#discussion_r1704667411

Reply:
Oops, `**` instead of `*`: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Sending a fix.